### PR TITLE
Jooq api

### PR DESCRIPTION
**Pull request by boazco**

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt [hunk 1]
```diff
@@ -4,9 +4,11 @@ import jakarta.ws.rs.BadRequestException
 import org.jooq.DSLContext
 import org.jooq.RecordMapper
 import org.jooq.Record
+import org.springframework.context.annotation.Bean
+import org.springframework.stereotype.Component
 import java.util.*
 
-
+@Component
 class OwnerDao(private val sql: DSLContext) {
```
Comment (boazco) [inline src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt:13]: class OwnerDao @autoWired(...)

File: src/main/kotlin/com/hibob/academy/resource/PetResource.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import java.awt.PageAttributes
+import java.util.*
+
+@Component
+@Path("/pets")
+class PetResource(
+    private val petService: PetService
+) {
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
```
Comment (boazco) [inline src/main/kotlin/com/hibob/academy/resource/PetResource.kt]: Move above

File: src/main/kotlin/com/hibob/academy/resource/ownerResource.kt [hunk 3]
```diff
@@ -1,42 +1,43 @@
...
     @Consumes(MediaType.APPLICATION_JSON)
-    fun updateOwnersPetType(@PathParam("ownerId") id: String, ownersPetType: String): Response {
-        if(fetchOwner(id) == null){
-            return Response.status(404).build()
+    fun createOwner(@RequestBody owner: Owner): Response {
+        val ownerId = ownerService.createOwner(owner)
+        if (ownerId != null) {
+            return Response.ok(ownerId).build()
+        } else {
+            return Response.status(Response.Status.BAD_REQUEST).build()
```
Comment (boazco) [inline src/main/kotlin/com/hibob/academy/resource/ownerResource.kt]: move to service

File: src/main/kotlin/com/hibob/academy/service/ownerService.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        return ownerDao.getOwners(companyId)
+    }
+
+    fun getOwnerByPetId(petId: UUID, companyId: Long): MutableList<Owner> {
+        return ownerDao.getOwnerByPetId(petId, companyId)
+    }
+
+    fun createOwner(owner: Owner): UUID? {
+        return ownerDao.createOwner(owner.companyId, owner.employeeId, owner.name)
+        //NEED TO CHANGE the dao create owner so it gets OwnerNoID (AFTER PREV PR APPROVED)
```
Comment (boazco) [inline src/main/kotlin/com/hibob/academy/service/ownerService.kt]: please rebase

File: src/main/kotlin/com/hibob/academy/resource/ownerResource.kt [hunk 5]
```diff
@@ -1,42 +1,43 @@
...
 
-    @PUT
-    @Path("/{ownerId}/updateType")
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
     @Consumes(MediaType.APPLICATION_JSON)
-    fun updateOwnersPetType(@PathParam("ownerId") id: String, ownersPetType: String): Response {
-        if(fetchOwner(id) == null){
-            return Response.status(404).build()
+    fun createOwner(@RequestBody owner: Owner): Response {
```
Comment (boazco) [inline src/main/kotlin/com/hibob/academy/resource/ownerResource.kt]: No Owner provided but different data class

File: src/main/kotlin/com/hibob/academy/resource/PetResource.kt [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    @Path("/byOwner/{companyId}/{ownerId}")
+    fun getPetsByOwnerId(@PathParam("companyId") ownerId: UUID, @PathParam("ownerId") companyId: Long): Response {
+        return Response.ok(petService.getPetsByOwnerId(ownerId, companyId)).build()
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun createPet(@RequestBody pet: Pet): Response {
```
Comment (boazco) [inline src/main/kotlin/com/hibob/academy/resource/PetResource.kt]: No Owner provided but different data class

File: src/test/kotlin/com/hibob/academy/resource/OwnerResourceTest.kt [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package com.hibob.academy.resource
+
+import org.junit.jupiter.api.Assertions.*
+
+class OwnerResourceTest(){
+
+}
```
Comment (boazco) [inline src/test/kotlin/com/hibob/academy/resource/OwnerResourceTest.kt]: can remove

**Review by omermelamed — APPROVED** [review]

**Pull request merged**